src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/AbstractRequest.java

```diff
@@ -0,0 +1,407 @@
+package com.minecolonies.coremod.colony.requestsystem.requests;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.colony.requestsystem.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.RequestState;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.blockout.Log;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Abstract skeleton implementation of a request.
+ *
+ * @param <R> The type of request this is.
+ */
+public abstract class AbstractRequest<R> implements IRequest<R>
+{
+
+    @NotNull
+    private final IToken       token;
+    @NotNull
+    private final R            requested;
+    @NotNull
+    private final List<IToken> children;
+    @NotNull
+    private final IRequester   requester;
+    @NotNull
+    private RequestState state = RequestState.CREATED;
+    @Nullable
+    private R      result;
+    @Nullable
+    private IToken parent;
+    @SuppressWarnings("squid:S1170")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1170' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1309)